pyop2/types/data_carrier.py

```diff
@@ -44,6 +44,9 @@ def cdim(self):
         the product of the dim tuple."""
         return self._cdim
 
+    def increment_dat_version(self):
```

Comment:
declare this abstract.

Reply:
The problem is that this requires all the subclasses to implement it and at the moment `Global`, `Dat` and `MixedDat` get the implementation of `increment_dat_version` via another route, i.e. `VecAccessMixin`. That’s because we don’t want to duplicate the object versioning setup in `Global` and `AbstractDat` objects so I put it in the appropriate common ancestor (`VecAccessMixin`).